pydantic/_internal/_core_utils.py

```diff
@@ -212,6 +212,7 @@ def _handle_other_schemas(self, schema: core_schema.CoreSchema, f: Walk) -> core
     def _handle_ser_schemas(self, ser_schema: core_schema.SerSchema, f: Walk) -> core_schema.SerSchema:
         schema: core_schema.CoreSchema | None = ser_schema.get('schema', None)
         if schema is not None:
+            ser_schema = ser_schema.copy()
```

Comment:
```suggestion
            ser_schema = smart_deepcopy(ser_schema)
```

Can we match the approach used in this PR? https://github.com/pydantic/pydantic/pull/9980

Reply:
I would rather stick with `.copy` here, as it is for the rest of the `_WalkSchema` implementation. `deepcopy` on a dict is slower and does not have any impact here, as `.copy` will create a shallow copy and this is enough (as we are only changing a top level key here, in this case the `'schema'` key).

Reply:
Sounds good!